spec/features/user_can_view_all_posts_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
+
+RSpec.feature "View all posts", type: :feature do
+  scenario "User can navigate to the all-posts-page via their welcome page" do
+    sign_up
+    click_link "View All Posts"
+    expect(current_path).to eq("/posts")
+  end
+
+  scenario "User can see the posts they have made on the all posts page" do
+    sign_up
+    click_link "View Your Posts"
+    click_link "New post"
+    fill_in("post_message", with: "Hello")
+    click_button "Submit"
+    click_link "View All Posts"
+    expect(current_path).to eq("/posts")
+    expect(page).to have_content("Hello")
+  end
+
+  scenario "User can see other user's posts as well as their own" do
+    sign_up("user@gmail.com")
+    click_link "View Your Posts"
```

Comment:
I think there is a `create_a_post` helper method in the web_helpers file which could make this more concise

Reply:
Nice one Danny - we've updated the tests with this